docs/index.md

```diff
@@ -7,77 +7,140 @@
 
-```
-pip uninstall llm-app-test 
-pip install llm-app-test==0.1.0b3.post3
-```
-#### Method 2: Force Reinstall (if Method 1 fails)
+## Testing Philosophy
 
-```
-pip install --force-reinstall llm-app-test==0.1.0b3.post3
-```
-#### Verification
-After rolling back, verify the installation:
-```
-import llm_app_test 
-print(llm_app_test.version) # Should show 0.1.0b3.post3
-```
+When integrating LLMs into your application, treat them as you would any closed-source third-party library:
 
-## Need testing ideas? Check out the tests we used to test llm-app-test [here](https://github.com/Shredmetal/llmtest/tree/main/tests)
+1. Write tests for expected behaviour
+2. Focus on interface boundaries
+3. Test application-level functionality
+4. Maintain clear separation of concerns
+
+### ⚠️ Important Information on Understanding Responsibilities
+
+This library is built by software engineers to give software engineers a tool to validate the behaviour of applications that have had an LLM stuffed in them. It is **NOT** a Data Science tool nor a replacement for model metrics used by Data Science teams to validate model suitability.
+
+#### Software Engineer's Role
+
+- Write tests for expected application behaviour
+- Validate inputs and outputs
+- Ensure proper integration
+- Monitor system performance
+- Escalate consistent failures to DS team (as this might indicate a fundamental problem with the model, or perhaps to seek assistance with the `expected_behavior` prompt in the `assert_behavioral_match` function)
+
+#### Data Science Team's Role
+
+- Handle model-level issues
+- Address consistent test failures
+- Evaluate model suitability
+- Optimise model performance
+- Adjust prompts when needed
+
+### When to Escalate
+
+Escalate to your Data Science team when:
+
+1. Tests consistently fail despite correct implementation
+2. Model responses are consistently inappropriate
+3. Performance degradation is observed
+4. Pattern of failures indicates model-level issues
+
+### 🔍 What Makes This Different?
+
+This is an ENGINEERING tool, not a data science tool. The difference is crucial:
+
+Data Science Tools:
+- Test model performance
+- Evaluate model accuracy
+- Measure model metrics
+
+llm_app_test (Engineering Tool):
+- Tests your APPLICATION code
+- Validates integration points
+- Ensures system behaviour
+- Maintains production reliability
+
+Think of it this way: You don't test Redis itself, you test your application's use of Redis. 
+Similarly, llm_app_test helps you test your application's use of LLMs.
+
+
+## In summary:
+
+### What llm_app_test Does
 
-## What llm_app_test Does
 - Tests LLM applications (not the LLMs themselves)
 - Validates system message + prompt template outputs
-- Ensures semantic equivalence of responses
+- Ensures expected behavioral patterns in responses
 - Tests the parts YOU control in your LLM application
 
-## What llm_app_test Doesn't Do
+### What llm_app_test Doesn't Do
```

Comment:
llm-app-test

Reply:
Undone - standardise to llm_app_test because that's what the import statement is going to say. llm-app-test is only used for pip install because PyPI converts underscores to hyphens